aquadoggo/src/db/stores/blob.rs

```diff
@@ -53,6 +55,98 @@ impl SqlStore {
         };
         document_to_blob_data(self, blob_document).await
     }
+
+    /// Purge blob data from the node _if_ it is not related to from another document.
+    pub async fn purge_blob(&self, document_id: &DocumentId) -> Result<(), SqlStoreError> {
+        // Collect the view id of any existing document views which contain a relation to the blob
+        // which is the purge target.
+        let blob_reverse_relations = reverse_relations(&self.pool, document_id, None).await?;
+
+        // If there are no documents referring to the blob then we continue with the purge.
+        if blob_reverse_relations.is_empty() {
+            // Collect the document view ids of all pieces this blob has ever referred to in it's
```

Comment:
Add a test to check purging does remove all pieces (even if the blob was updated).

Reply:
We don't _need_ to support blob updating, but so far I think we do.